### PR TITLE
Add support for per-cell data types in Grid Excel exports

### PR DESCRIPTION
**Pull request by ghsolomon**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by ghsolomon** [comment]
See related PR in Hoist-Core: https://github.com/xh/hoist-core/pull/265

File: svc/GridExportService.js [hunk 1]
```diff
@@ -381,6 +386,23 @@ export class GridExportService extends HoistService {
...
+        if (isInteger(v))   return ifTypeNot(NUMBER, INT);
+        if (isNumber(v))    return ifTypeNot(NUMBER);
+        if (isLocalDate(v)) return ifTypeNot(LOCAL_DATE);
+        if (isDate(v))      return ifTypeNot(DATE);
+        if (isString(v))    return ifTypeNot(PWD, STRING, AUTO);
+        if (isArray(v))     return ifTypeNot(TAGS, JSON, AUTO);
+        if (isObject(v))    return ifTypeNot(JSON, AUTO);
+
+        return null;
+    }
```
Comment (TomTirapani) [inline svc/GridExportService.js:405]: I'm not clear on the circumstances when a cell's type would be different from the store field type? Isn't this only really possible if the store field type is AUTO?

File: cmp/grid/Grid.js [hunk 2]
```diff
@@ -781,12 +781,14 @@ class GridLocalModel extends HoistModel {
...
 
-        return XH.gridExportService.getExportableValueForCell({
+        const exportValue = XH.gridExportService.getExportableValueForCell({
             gridModel: model,
             record,
             column: xhColumn,
             node
         });
+
+        return exportValue?.value ?? exportValue;
```
Comment (TomTirapani) [inline cmp/grid/Grid.js:791]: Good catch!

Comment (ghsolomon) [reply cmp/grid/Grid.js:791]: Looks like CSV export also affected by this.  Making another tweak.

https://github.com/xh/hoist-react/issues/2797

**Pull request merged**